source/mod_ffield_Table_n_Track.f90

```diff
@@ -0,0 +1,1304 @@
+  subroutine Tload(this,norm,iErr)
+    ! Mod from SixTrack
+    ! ---------------------------------------------------------------------------------------------- !
+    use mod_units, only : f_requestUnit
+    use crcoall,   only : lout
+
+    implicit none
+
+    ! interface variables
+    ! ---------------------------------------------------------------------------------------------- !
+    class(ffTable_n_Track), intent(inout) :: this
+    logical,                intent(inout) :: iErr
+    real(kind=fPrec),       intent(in)    :: norm      ! Normalization
+
+    integer :: lun
+    
+
+    if (this%chk_Status<1) then
+      write(lout,"(a,i1)") "FFIELD> ERROR while loading file '"//trim(this%ffFNames)//"' -> status ",this%chk_Status
+      iErr = .true.
+      return
+
+    else if (this%chk_Status == 1) then
+      ! Request to find unit
+      ! -------------------------------------------------------------------------------------------- !
+      call f_requestUnit(this%ffFNames,lun)
+
+      ! Detect the mamixum exposant for x and y, and the number of point in z
+      ! -------------------------------------------------------------------------------------------- !
+      call ReadExpMax(this,lun,iErr)
+      if (iErr) return
+
+      ! Generate the vecto potential table
+      ! -------------------------------------------------------------------------------------------- !
+      call ReadVectPotCoeff(this,lun,norm,iErr)
+      if (iErr) return
+
+      this%s=this%s-1
+
+      this%chk_Status=2
+
+    end if
+  end subroutine Tload
+
+
+  subroutine ReadExpMax(this,lun,iErr)
+    ! Mod from SixTrack
+    ! ---------------------------------------------------------------------------------------------- !
+    use numerical_constants, only : zero, c1e12, c1m12
+    use crcoall,             only : lout
+    use parpro,              only : mInputLn
+    use mod_units,           only : f_open, f_close
+    use string_tools,        only : chr_split, chr_cast
+
+    implicit none
+
+    ! interface variables
+    ! ---------------------------------------------------------------------------------------------- !
+    class(ffTable_n_Track), intent(inout) :: this
+    integer,                intent(inout) :: lun  ! Unit for file to be open
+    logical,                intent(inout) :: iErr ! Error return
+
+    character(len=mInputLn) :: inLine             ! Line of the file
+    integer :: nSplit                             ! Nb element in line
+    logical :: ffErr                              ! Error returned
+    integer :: istat                              ! Check. accecibility of the file
+    integer :: line                               ! Nb of line in file
+    integer :: n,m,s                              ! Exposant max for x and y, Nb of point in z
+    integer(kind=int32) :: expx,expy,expz         ! Expo. for x, y and z
+    real(kind=fPrec) :: st, sm1                   ! Parametter for the detection of new step in z
```

Comment:
Why int32? integer generally defaults to 32 bit, and in the event that it doesn't, this will cause warnings everywhere where there is a mismatch.

We generally just use integer at default size unless there is a specific need for either int16 or int64, which there is a few places.

It's not a big issue, it just seems a bit unnecessary.